src/NuGet.Server/Infrastructure/ServerPackageRepository.cs

```diff
@@ -539,6 +542,7 @@ public void ClearCache()
 
                 _serverPackageStore.Clear();
                 _serverPackageStore.Persist();
+                _needsRebuild = true;
```

Comment:
This method should be done in a ``lock (_syncLock)`` also.  Both the get and create are locked, it makes sense that clear should be also. There could be a race case otherwise since a clear can start from a URL request.